src/webui/components/AutoComplete/index.js

```diff
@@ -91,12 +115,16 @@ const AutoComplete = ({
           color,
           onKeyDown,
         }}
-        renderSuggestionsContainer={options => (
-          <Paper {...options.containerProps} square>
-            {options.children}
+        renderSuggestionsContainer={({ containerProps, children, query }) => (
+          <Paper {...containerProps} square>
+            {suggestionsLoaded && children === null && query !== '' && renderMessage(SUGGESTIONS_RESPONSE.NO_RESULT)}
```

Comment:
I would change it...if the query is !== '"' means query === true ....so you can remove it... I would also use !! for the children...but it's only a tip 😉 suggestionsLoaded && !children && query &&

Reply:
It handles a very special case here:
1. Loaded, Loading - If you keep like this it makes things simple. Easy to understand and read.
2. In our search component when you focus out `children` becomes `null`. So the situation is like this.

You have `pnpm` in the search box. (it means `loading === false`) and your are also focused out, that is`children === null` and `query === 'pnpm'`.   So you see `no results found` always.

Reply:
-> Loaded, Loading - If you keep like this it makes things simple. Easy to understand and read.

yes, it is easier to understand, but I still don't agree on that..on one side it's simple to understand, on another side we have to deal with one more state, which also make it less simple...I try to reuse as much as I can...what do you think @juanpicado ?

Reply:
@ayusharma fine ... but I would still do it another way ...

Reply:
If you solve this problem:

You have pnpm in the search box. (it means loading === false) and you are also focused out, that is children === null and query === 'pnpm'. So you see `no results found` always.

so you show `No results found.` with the following condition
suggestionsLoading === false && !children && query && ...

so, (onBlur) let's check values:
suggestionsLoafing = false (api has loaded the data)
children = null
query = 'pnpm'

if you keep above values in the equation:

false === false && !null && 'pnpm' && ...

i.e. true && true && true && ...

Reply:
My thoughts on this are a bit divided. 

* First of all, my concern with loading. I see this logic of `loaded`, `loading` and `error` being handled in each component that hits the API. Eg: sidebar (https://github.com/verdaccio/verdaccio/blob/master/src/webui/components/PackageSidebar/index.jsx#L39). We need a centralized point to handle all these states instead each component. I hope you see the problem here.
*  About the logic @priscilawebdev is corcerned, I think that component handle the logic a bit weird, I've been debugging and I think we need something else than 1 state. But not 100%.
![screen shot 2018-11-23 at 8 01 34 am](https://user-images.githubusercontent.com/558752/48931393-575edb80-eef6-11e8-9f0e-d036ce77bf7f.png)
* I'd suggest let it as it is until we found a way to centralise this logic in ALL components. 

We were agree in some point we will move to some central state, so, I'd suggest discuss this in that moment and refactor all API calls at that point.